staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -258,7 +293,14 @@ func ValidateCustomResourceConversion(conversion *apiextensions.CustomResourceCo
 
 // ValidateCustomResourceDefinitionSpecUpdate statically validates
 func ValidateCustomResourceDefinitionSpecUpdate(spec, oldSpec *apiextensions.CustomResourceDefinitionSpec, established bool, fldPath *field.Path) field.ErrorList {
-	allErrs := ValidateCustomResourceDefinitionSpec(spec, fldPath)
+	requireRecognizedVersion := true
+	if oldSpec.Conversion != nil {
+		// only skip requiring recognized versions if we had an unrecognized version, not for any other invalid values.
+		wasValidExceptInvalidVersions := len(validateConversionReviewVersions(oldSpec.Conversion.ConversionReviewVersions, false, fldPath)) == 0
```

Comment:
same comment about checking for an accepted version directly instead of calling validation twice and inferring from the different results